scripts/release/marketplaceRelease.js

```diff
@@ -0,0 +1,135 @@
+        if (!downloadUrl) {
+            throw new Error("Could not retrieve MPK url from GitHub tag");
+        }
+        return downloadUrl;
+    } else {
+        throw new Error("Could not retrieve release tag information from GitHub");
+    }
+}
+
+async function createDraft(marketplaceId, version, studioProMinimumVersion) {
+    console.log(`Creating draft in the Mendix Marketplace..`);
+    const url = `${config.appStoreUrl}/packages/${marketplaceId}/version`;
+    try {
+        const [major, minor, patch] = version.split(".");
+        const assetUrl = await getGithubAssetUrl();
+        const body = {
+            VersionMajor: major ?? 1,
+            VersionMinor: minor ?? 0,
+            VersionPatch: patch ?? 0,
+            StudioProVersion: studioProMinimumVersion,
+            IsSourceGitHub: "true",
+            GithubRepo: {
+                UseReadmeForDoc: false,
+                ArtifactURL: assetUrl
+            }
+        };
+
+        return fetch("POST", url, JSON.stringify(body));
+    } catch (error) {
+        error.message = `Failed creating draft in the appstore with error: ${error.message}`;
+        throw error;
+    }
+}
+
+function publishDraft(UUID) {
+    console.log(`Publishing draft in the Mendix Marketplace..`);
+    const url = `${config.appStoreUrl}/version/${UUID}/publish`;
+    try {
+        return fetch("PUT", url);
+    } catch (error) {
+        error.message = `Failed publishing draft in the appstore with error: ${error.message}`;
+        throw error;
+    }
+}
+
+async function fetch(method, url, body) {
+    let response;
+    const httpsOptions = {
+        method,
+        redirect: "follow",
+        headers: {
+            Accept: "application/json",
+            "Mendix-Username": process.env.MARKETPLACE_USERNAME,
+            "Mendix-ApiKey": process.env.MARKETPLACE_API_KEY
+        },
+        body: undefined
+    };
+
+    if (body) {
+        httpsOptions.body = body;
+        httpsOptions.headers["Content-Type"] = "application/json";
+    }
+
+    console.log(`Fetching URL (${method}): ${url}`);
+    try {
+        response = await nodefetch(url, httpsOptions);
+    } catch (error) {
+        console.error(error);
+    }
+    console.log(`Response status Code ${response.status}`);
+    if (response.status === 409) {
+        throw new Error(
+            `Fetching Failed (Code ${response.status}). Possible solution: Check & delete drafts in Mendix Marketplace.`
+        );
+    } else if (response.status === 503) {
+        throw new Error(`Fetching Failed. "${url}" is unreachable (Code ${response.status}).`);
+    } else if (response.status !== 200 && response.status !== 201) {
+        throw new Error(`Fetching Failed (Code ${response.status}). ${response.statusText}`);
+    } else if (response.ok) {
+        return response.json();
+    } else {
+        throw new Error(`Fetching Failed. Unhandled status code: ${response.status}`);
```

Comment:
🎄  (nice to have): supply the status text as well.

Reply:
Fixed